Hack Specific Registration Page

miguelaenlle opened the pull request:
Here is the Hack Specific Page:

![image](https://github.com/user-attachments/assets/ac2d7b9a-71b7-4c1c-b454-d210f00aa8ba)


I also created a `QuestionHeader` component that has an `h3` tag and `required` asterisk for questions on the Hack Specific Page:

![image](https://github.com/user-attachments/assets/ef32cd00-60b9-46cf-a27c-7f66ed47c595)

![image](https://github.com/user-attachments/assets/5d399788-ed52-4298-b6c0-5bf8521c711c)

Let me know if this is okay, or if I should replace this component with `h3` tags/`required` asterisks, like this:

![image](https://github.com/user-attachments/assets/0d3b5c22-9f6a-43e3-82e7-9b3f474dcbe0)

miguelaenlle committed:
Implemented the HackSpecific page

miguelaenlle committed:
Reformatted the option values and TextInput component names

miguelaenlle committed:
Uncommented Education/Transportation buttons for Hack Specific

miguelaenlle committed:
Removed unnecessary imports in QuestionHeader

J164 commented on components/Registration/Registration.tsx:
```diff
@@ -45,14 +47,13 @@ const Form: React.FC = () => {
-                            handleNext={nextPage}
-                        />
-                    ))}
+                {formIndex != pages.length - 1 && true && (
```
the "true" can be removed now that we have more than one page. It was only for testing :)

J164 commented on components/QuestionHeader/QuestionHeader.tsx:
```diff
@@ -0,0 +1,18 @@
+    return (
+        <h3>
+            {text}
+            {required && <span className={styles.required}>*</span>}
```
As @BeckyBlake pointed out, the asterisks are the same color as the question text in the Figma which was an oversight on my part. For now we can just go back to using <h3>

J164 commented on components/Registration/Pages/HackSpecific/HackSpecific.tsx:
```diff
@@ -0,0 +1,154 @@
+                    text="In a couple of sentences, please explain why you are interested in participating in HackIllinois 2024."
+                    required
+                />
+                <TextInput
```
It might be nice for this input field to be taller in the form. Is this something that can be done by applying a className or will changes need to be made to TextInput?

J164 reviewed (changes requested):
Great start! I left a few comments to address before we merge.

miguelaenlle committed:
Removed question header; removed the true; extended textarea height

miguelaenlle commented:
@J164 , I have implemented the changes.

I adjusted the TextInput height using a custom className passed into TextInput, about 6 lines tall. Here's what it looks like:

<img width="608" alt="image" src="https://github.com/user-attachments/assets/5e55dbd1-0000-4a53-99d1-0f6166a29d4c">

J164 reviewed (approved):
Looks great!
